web/packages/design/src/Popover/Transition.tsx

```diff
@@ -16,18 +16,32 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React, { useLayoutEffect } from 'react';
+import React, { RefObject, useLayoutEffect } from 'react';
 
+import { useResizeObserver } from 'design/utils/useResizeObserver';
+
+/**
+ * Transition is a helper for firing certain effects from Popover, as it's way easier to use them
+ * this way than integrating with the component lifecycle.
+ */
 export function Transition({
   onEntering,
+  enablePaperResizeObserver,
+  paperRef,
+  onPaperResize,
   children,
 }: React.PropsWithChildren<{
   onEntering: () => void;
+  enablePaperResizeObserver: boolean | undefined;
+  paperRef: RefObject<HTMLElement>;
+  onPaperResize: () => void;
 }>) {
   // Note: useLayoutEffect to prevent flickering improperly positioned popovers.
   // It's especially noticeable on Safari.
-  useLayoutEffect(() => {
-    onEntering();
-  }, []);
+  useLayoutEffect(onEntering, [onEntering]);
```

Comment:
Hmm, true. I knew that the callback passed by Popover is not going to change and I was trying to follow rules around useEffect. But this is one of the rare cases where might want to run something only on mount.

Though, one could argue that this callsite should follow useEffect rules and the caller should make sure that `onEntering` won't change. The problem in general seems that this code is very DOM/lifecycle oriented instead of being about menu opening / menu closing. And on top of that there's the fact that `Transition` is just a workaround for `Popover` being a class component.